### PR TITLE
bqn-interactive: abstract constants repl functions

### PR DESCRIPTION
**Pull request by doyougnu**
Hello,

This looks like a cool language so I thought I would contribute. This PR does the follow:

1. abstract constants out to `defcustom` so users can override them if desired
2. add a bunch of convenience functions for interacting with the `BQN` repl
3. adds `autoload` to `bqn-run`

Please let me know if any edits are necessary.

### CODE REVIEW

File: bqn-interactive.el [hunk 1]
```diff
@@ -51,25 +51,102 @@
     map)
   "Keymap for keymap mode buffers.")
 
+(defcustom bqn--process-name "BQN"
```
[the remark this answers is not in the document]
Comment (doyougnu) [reply bqn-interactive.el]: Totally agree, in fact I think it is a mistake to use `defcustom` here, this is the emacs process name for the bqn process so I think it should be implied to be private via double-dash. Resolved in cd4262e841cf7f5e579ac17fe33de3f817a1f4d7

File: bqn-interactive.el [hunk 2]
```diff
@@ -51,25 +51,102 @@
...
+      (apply 'make-comint-in-buffer
+             bqn--process-name
+             buffer
 	     bqn-program bqn-cli-arguments)
-      (switch-to-buffer-other-window "*BQN*")
+      (switch-to-buffer-other-window *bqn-process-buffer-name*)
       (bqn-inferior-mode)
       (set-input-method "BQN-Z"))))
 
+(defun bqn-process-execute-region-and-follow (start end &optional dont-follow)
```
[the remark this answers is not in the document]
Comment (doyougnu) [reply bqn-interactive.el]: Great points. Thank you for the review.  Resolved in 0e5d580.

I've left `bqn-process-execute-line-and-follow` and `bqn-process-execute-buffer-and-follow` as is, and removed `bqn-process-execute-region-and-follow`, please let me know if any further revision is necessary. I'm happy to contribute

File: bqn-interactive.el [hunk 3]
```diff
@@ -51,25 +51,102 @@
...
+   process buffer"
+  (interactive)
+  (bqn-process-execute-region (point-at-bol) (point-at-eol)))
+
+(defun bqn-process-execute-buffer-and-follow ()
+  "Send the current buffer to BQN process and focus BQN process buffer"
+  (interactive)
+  (bqn-process-execute-region (point-min) (point-max)))
+
+(defun bqn-process-execute-region (start end)
```
[the remark this answers is not in the document]
Comment (doyougnu) [reply bqn-interactive.el]: See my comments here: https://github.com/AndersonTorres/bqn-mode/pull/7#discussion_r739479004

File: bqn-interactive.el [hunk 4]
```diff
@@ -102,8 +100,9 @@
       (bqn-inferior-mode)
       (set-input-method "BQN-Z"))))
 
-(defun bqn-process-execute-region-and-follow (start end &optional dont-follow)
-  "Send the current region to the bqn-process-session"
+(defun bqn-process-execute-region (start end &optional dont-follow)
+  "Send the current region to the bqn-process-session, when dont-follow is
+   non-nil maintain focus on buffer function was called from."
```
Comment (AndersonTorres) [inline bqn-interactive.el:105]:
```suggestion
  "Send the current region to the bqn-process-session.
When DONT-FOLLOW is non-nil, maintain focus on the buffer where the function was called from."
```

In Elisp docstrings, the variables are cited in CAPSLOCK.

https://www.emacswiki.org/emacs/DocString

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**